misc/python/materialize/cli/run.py

```diff
@@ -140,28 +140,31 @@ def main() -> int:
         del args.args[0]
 
     if args.program in KNOWN_PROGRAMS:
-        build_retcode = _build(args, extra_programs=[args.program])
+        (build_retcode, built_programs) = _build(args, extra_programs=[args.program])
         if args.build_only:
             return build_retcode
 
         if args.release:
-            path = MZ_ROOT / "target" / "release" / args.program
+            artifact_path = MZ_ROOT / "target" / "release"
         else:
-            path = MZ_ROOT / "target" / "debug" / args.program
+            artifact_path = MZ_ROOT / "target" / "debug"
+        program_path = artifact_path / args.program
 
         if args.disable_mac_codesigning:
             if sys.platform != "darwin":
                 print("Ignoring --disable-mac-codesigning since we're not on macOS")
             else:
                 print("Disabled macOS Codesigning")
         elif sys.platform == "darwin":
-            _macos_codesign(path)
+            for program in built_programs:
+                path = artifact_path / program
+                _macos_codesign(str(path))
```

Comment:
Do you also need `_macos_codesign(str(program_path))`?

Reply:
`args.program` is part of the `builtin_programs` list, that's definitely not obvious though. I removed the `program_path` variable altogether, does that makes things a little more clear?